flang/lib/Optimizer/CodeGen/TargetRewrite.cpp

```diff
@@ -575,16 +574,13 @@ class TargetRewrite : public TargetRewriteBase<TargetRewrite> {
             rewriter->setInsertionPointToStart(&func.front());
             auto cplxTy = oldArgTys[fixup.index - offset - fixup.second];
             auto undef = rewriter->create<UndefOp>(loc, cplxTy);
-            auto iTy = rewriter->getIntegerType(32);
-            mlir::Value zero = rewriter->create<mlir::arith::ConstantOp>(
-                loc, iTy, mlir::IntegerAttr::get(iTy, 0));
-            mlir::Value one = rewriter->create<mlir::arith::ConstantOp>(
-                loc, iTy, mlir::IntegerAttr::get(iTy, 1));
+            auto zero = rewriter->getIntegerAttr(rewriter->getIndexType(), 0);
```

Comment:
Use iTy/getIntegerType(32) instead of indextype to make it compatible with previous code.

Reply:
There is a downstream constraint that must be met here. LLVM requires 32 bit integers in some instructions. It will crash (or assert) otherwise.